### PR TITLE
Enhancement: Allow setting breakpoints when two sources have the same name

### PR DESCRIPTION
**Pull request by haltman-at**
Addresses #4049.  Or at least partly does?  It's something of a stopgap, but I don't see how we're going to do better for now.

Currently, when you try in the CLI debugger to add a breakpoint in a specified file, if there are multiple matching files, the CLI asks for disambiguation.  However, as per #4049, sometimes some sources might be named identically.  So this PR makes it so that if there's one matching file whose source path is a substring of all other matching source paths -- including, say, if there are a bunch of source paths which are identical -- then instead of asking for disambiguation, it'll print a warning and then add (or remove) them all.

In order to do this, I factored the breakpoint-setting function a bit, but be warned, it's still a big mess of imperative code that prints to the console a bunch!  Changing that is perhaps something that can be undertaken at a later time.  Anyway, now there's one function that determines the breakpoint objects to add/remove, and another to perform resolution and do the adding.  (They both print to the console...).  The former function returns the list of breakpoints, which will usually have length 1, but it can also return multiple breakpoints in the case discussed above, can return `[]` if something goes wrong and it doesn't know where to add anything, and will return `null` in the special case of `B all` (or `b all` :P ).  So yeah I used `[]` to mean "no breakpoints" and `null` to mean "all breakpoints"... this is called out in the comments so hopefully that isn't confusing.

Note I didn't really test the actual case this PR is made for; but uh I did test it on existing cases to make sure it doesn't break things?

Also, while I was at it, I made two other changes:
1. I fixed a bug I accidentally introduced in #4030, where typing `b` could get you an error if the debugger can't determine your location
2. I made it so that typing `b` followed by a large negative number won't hang the debugger!

### CODE REVIEW

File: packages/core/lib/debug/interpreter.js [hunk 1]
```diff
@@ -132,42 +152,76 @@ class DebugInterpreter {
...
+            shortSource => matchingSources.every(
+              source =>
+                typeof source.sourcePath !== "string" //just ignore these I guess?
+                || source.sourcePath.includes(shortSource.sourcePath)
+            )
+          )
+        ) {
+          //exceptional case
+          this.printer.print(
+            `WARNING: Multiple source files matched, but one is a proper substring of all others, so disambiguation is not possible.  Acting on all of them.`
```
Comment (gnidan) [inline packages/core/lib/debug/interpreter.js:172]: I think this can be simplified:
```suggestion
            `WARNING: Multiple source files matched; could not disambiguate.  Setting breakpoints for:`
```

Comment (haltman-at) [reply packages/core/lib/debug/interpreter.js:172]: Well, we can't go with this exactly.  It can't be "setting breakpoints for", because this also applies to *removing* breakpoints.  We can have it switch as appropriate (I didn't bother as you can see, but it wouldn't be hard), but it can't just be "setting".

Also, I don't like "setting breakpoints for:" because the normal messages will still print for each.  If it were "setting breakpoints for:", you would expect to just see the *location* printed for each, not the full message.

Also, I think we do need a "but" in there.  "Multiple source files matched; could not disambiguate" sounds like we don't perform disambiguation.  We need to say "Multiple source files matced, but could not disambiguate" to make it clear that we would disambiguate if we could.

...although, really, *we* don't disambiguate, we ask the *user* to disambiguate.  Hence why I said "disambiguation is not possible", not "could not disambiguate".  IDK, maybe that's getting too nitpicky.

Comment (cds-amal) [reply packages/core/lib/debug/interpreter.js:172]: Does the user need to know the internals? Can we inform the user we encountered ambiguity because of multiple source name collision and set expectations that way?

Comment (cds-amal) [reply packages/core/lib/debug/interpreter.js:172]: Thanks for the clarification, it makes sense. How about rewording to present course of action followed by reasoning? 

> WARNING: acting on all sources because disambiguation is not possible due to one filename is a proper substring of all others.

Comment (haltman-at) [reply packages/core/lib/debug/interpreter.js:172]: So, rewording a little, if we take @cds-amal's approach, but also shorten things a bit as @gnidan wanted, I might write something like...

> WARNING: Acting on all matching sources because disambiguation between them is not possible.

How does that sound? :)

Comment (cds-amal) [reply packages/core/lib/debug/interpreter.js:172]: I'm ok with that!

Comment (haltman-at) [reply packages/core/lib/debug/interpreter.js]: OK, I've made this change.

**Review by cds-amal — APPROVED** [review]
Looks good!
